src/App.js

```diff
@@ -14,11 +15,126 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const filterOptions = Array.from(Array(10).keys());
 
-export default App;
+export class App extends React.Component {
+  state = {
+    isContentVisible: false,
+    isReversed: false,
+    sortedBy: '',
+    lengthFilter: '1',
+  };
+
+  enableContent = (event) => {
+    const button = event.target;
+
+    button.hidden = true;
+    this.setState({ isContentVisible: true });
+  }
+
+  sortAlphabetically = () => {
+    this.setState({
+      isReversed: false,
+      sortedBy: 'name',
+    });
+  }
+
+  sortByLength = () => {
+    this.setState({
+      isReversed: false,
+      sortedBy: 'length',
+    });
+  }
+
+  handleFilter = (event) => {
+    const { value } = event.target;
+
+    this.setState({
+      lengthFilter: value,
+    });
+  }
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortedBy: '',
+      lengthFilter: '1',
+    });
+  }
+
+  render() {
+    const {
+      isContentVisible, isReversed, sortedBy, lengthFilter,
+    } = this.state;
+
+    return (
+      <div className="App">
+        <h1>Goods</h1>
+
+        <button
+          type="button"
+          onClick={this.enableContent}
+        >
+          Start
+        </button>
+
+        <button
+          type="button"
+          disabled={!isContentVisible}
+          onClick={() => {
+            this.setState(state => ({ isReversed: !state.isReversed }));
+          }}
+        >
+          Reverse
+        </button>
+
+        <button
+          type="button"
+          disabled={!isContentVisible}
+          onClick={this.sortAlphabetically}
+        >
+          Sort by name
+        </button>
+
+        <button
+          type="button"
+          disabled={!isContentVisible}
+          onClick={this.sortByLength}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          disabled={!isContentVisible}
+          onClick={this.reset}
+        >
+          Reset
+        </button>
+
+        <select
+          disabled={!isContentVisible}
+          name="lengthFilter"
+          value={lengthFilter}
+          onChange={this.handleFilter}
+        >
+          {filterOptions.map(number => (
+            <option value={`${number + 1}`} key={number + 1}>
+              {number + 1}
+            </option>
+          ))}
+        </select>
+
+        {isContentVisible
+          && (
+            <GoodsList
+              goods={goodsFromServer}
+              reversed={isReversed}
+              sortedBy={sortedBy}
+              minLength={lengthFilter}
```

Comment:
Let's pass only prepared goods to the component GoodsList. It should only render the list, move all your operations (sorting, reversing etc) to the parent element and pass here only ready goods